elasticsearch_dsl/search_base.py

```diff
@@ -949,7 +949,7 @@ def to_dict(self, count: bool = False, **kwargs: Any) -> Dict[str, Any]:
         d = {}
 
         if self.query:
-            d["query"] = self.query.to_dict()
+            d["query"] = recursive_to_dict(self.query)
```

Comment:
The idea with these new recursive conversions to dict is that we want to get rid of all the internal classes that we can now use for attributes and sub-attributes.